app/models/assessment.rb

```diff
@@ -13,8 +13,9 @@ class Assessment < ApplicationRecord
   validates_presence_of :short_identifier
   validate :short_identifier_unchanged, on: :update
   validates_presence_of :description
-  validates_presence_of :due_date
+  validates_presence_of :due_date, if: ->(assessment) { assessment.type == 'Assignment' }
```

Comment:
I just realized that you can move this validation into the `Assignment` model rather than keep it here. That way you should be able to remove the `if`.